src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/PooledStringBuilder.cs

```diff
@@ -14,4 +14,19 @@ public static PooledDisposer<PooledStringBuilder> GetInstance(out StringBuilder
         instance = pooledInstance;
         return new PooledDisposer<PooledStringBuilder>(pooledInstance);
     }
+
+    void IPooled.Free(bool discardLargeInstances)
```

Comment:
Trying to understand how I would use a large instance StringBuilder. I assume I would need to do what is done in ArrayBuilder, add an ObjectPool to this class and then add a GetInstance method that potentially pulls from that pool?

Reply:
`PooledStringBuilder.GetInstance(out var sb)`

You can now have this new behavior with:

`PooledStringBuilder.GetInstance(discardLargeInstance: false, out var sb)`